File: javalin-openapi-processor/src/main/java/net/dzikoysk/openapi/processor/annotations/OpenApiParamInstance.java
package net.dzikoysk.openapi.processor.annotations;

import io.javalin.plugin.openapi.annotations.OpenApiParam;
import net.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper;

import javax.lang.model.element.AnnotationMirror;
import java.lang.annotation.Annotation;

@SuppressWarnings("ClassExplicitlyAnnotation")
final class OpenApiParamInstance extends AnnotationMirrorMapper implements OpenApiParam {

    public OpenApiParamInstance(AnnotationMirror mirror) {
        super(mirror);
    }

    @Override
    public boolean allowEmptyValue() {
        return getBoolean("allowEmptyValue");
    }

    @Override
    public boolean deprecated() {
        return getBoolean("boolean");
    }

    @Override
    public String description() {
        return getString("description");
    }

    @Override
    public boolean isRepeatable() {
        return getBoolean("isRepeatable");
    }

    @Override
    public String name() {
        return getString("name");
    }

    @Override
    public boolean required() {
        return getBoolean("required");
    }

    @Override
    public Class<?> type() {
        return null;
    }

    @Override
    public Class<? extends Annotation> annotationType() {
        return OpenApiParam.class;
    }

}

File: javalin-openapi-processor/src/main/kotlin/net/dzikoysk/openapi/processor/annotations/OpenApiParamInstance.kt
package net.dzikoysk.openapi.processor.annotations

import io.javalin.plugin.openapi.annotations.OpenApiParam
import net.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper
import javax.lang.model.element.AnnotationMirror
import javax.lang.model.type.TypeMirror

class OpenApiParamInstance(mirror: AnnotationMirror) : AnnotationMirrorMapper(mirror) {

    enum class In {
        QUERY, HEADER, PATH, COOKIE
    }

    fun allowEmptyValue(): Boolean =
        getBoolean("allowEmptyValue")

    fun deprecated(): Boolean =
        getBoolean("deprecated")

    fun description(): String =
        getString("description")

    fun isRepeatable(): Boolean =
        getBoolean("isRepeatable")

    fun name(): String =
        getString("name")

    fun required(): Boolean =
        getBoolean("required")

    fun type(): TypeMirror =
        getType("type")

    fun annotationType(): Class<out Annotation> =
        OpenApiParam::class.java

}